### PR TITLE
[CEKit maintenance] Start to version every module of jboss-eap-7-image repo

### PR DESCRIPTION
**Pull request by iankko**
which is now required to be able to build the image using CEKit v3.3.1.
See links below for more details:

* https://cekit.io/blog/2019/07/cekit-3.3.1-released/
* https://docs.cekit.io/en/latest/descriptor/module.html#version

Without this change it's not possible to build the images using CEKit 3.3.1 version (former versions work without issues).

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for other issues
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by iankko** [comment]
@luck3y @rgdoliveira PTAL (once got a chance)

@luck3y @rgdoliveira Yet one request, if this change is OK to go to `sso-cd-dev` branch. Could you possibly cherrypick it also into the [_EAP_CD_17_CR1_ tag](https://github.com/jboss-container-images/jboss-eap-7-image/releases/tag/EAP_CD_17_CR1)?

@goldmann jFYI

Thanks!, Jan

**Comment by luck3y** [comment]
@iankko Thanks Jan, I hadn't tried building with the new cekit yet. We're pretty far into CD17 at this point, so we're blockers only, I'll merge this and make a new tag for you if you like though, what would you like it to be called?

**Review by luck3y — APPROVED** [review]

**Comment by iankko** [comment]
> @iankko Thanks Jan, I hadn't tried building with the new cekit yet. We're pretty far into CD17 at this point, so we're blockers only, I'll merge this and make a new tag for you if you like though, what would you like it to be called?

Thanks for looking into this, Ken!

Yes, without this change CEKit 3.3.1 (tried `cekit-3.3.1-1.fc30.noarch`, but doesn't matter if its Fedora installation, or RHEL-7 one), returns error like the following:

```
[iankko@localhost redhat-sso-7-image]$ rpm -q cekit
cekit-3.3.1-1.fc30.noarch
```

```
[iankko@localhost redhat-sso-7-image]$ cekit -v build docker
...
2019-07-23 08:04:18,543 cekit        DEBUG    Reading descriptor from '/tmp/github/redhat-sso-7-image/target/repo/modules/sso/module.yaml' file...
validation.invalid
 --- All found errors ---
["Cannot find required key 'version'. Path: ''"]
2019-07-23 08:04:18,544 cekit        DEBUG    Schema validation failed: <SchemaError: error code 2: Schema validation failed:
 - Cannot find required key 'version'. Path: ''.: Path: '/'>
2019-07-23 08:04:18,544 cekit        ERROR    Cannot validate schema: Module
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/cekit/cli.py", line 334, in run
    command.execute()
  File "/usr/lib/python3.7/site-packages/cekit/builder.py", line 53, in execute
    self.before_generate()
  File "/usr/lib/python3.7/site-packages/cekit/builder.py", line 95, in before_generate
    self.generator.init()
  File "/usr/lib/python3.7/site-packages/cekit/generator/base.py", line 101, in init
    self.build_module_registry()
  File "/usr/lib/python3.7/site-packages/cekit/generator/base.py", line 159, in build_module_registry
    self.load_repository(os.path.join(base_dir, repo.target_file_name()))
  File "/usr/lib/python3.7/site-packages/cekit/generator/base.py", line 170, in load_repository
    os.path.dirname(module_descriptor_path))
  File "/usr/lib/python3.7/site-packages/cekit/descriptor/module.py", line 18, in __init__
    super(Image, self).__init__(descriptor)
  File "/usr/lib/python3.7/site-packages/cekit/descriptor/base.py", line 42, in __init__
    self.__validate()
  File "/usr/lib/python3.7/site-packages/cekit/descriptor/base.py", line 55, in __validate
    raise CekitError("Cannot validate schema: {}".format(self.__class__.__name__))
cekit.errors.CekitError: Cannot validate schema: Module
```

Obviously, as seen above, we also have unversioned modules. But adding versions just to our modules isn't enough, also versioning of parent modules we (re-)use is necessary. That's why I created this PR.

Yet, we need CEKit 3.3.X version, not to need manually to cherrypick / apply the [_microdnf_ workaround](https://github.com/cekit/cekit/pull/560) against former CEKit 3.2.X versions on each build / testing node, the image could be build / tested on. Hope this clarifies.

Regarding the cherrypicking of this change back to [EAP_CD_17_CR1](https://github.com/jboss-container-images/jboss-eap-7-image/releases/tag/EAP_CD_17_CR1) - actually, taking it back. I don't think we need it.

For the reproducible image builds, we just need to stick to some commit SHA (rather than using moving `master` target). But since [there's no difference in `master / sso-cd-dev` branch vs `EAP_CD_17_CR1` yet AFAICT](https://github.com/jboss-container-images/jboss-eap-7-image/compare/eap-cd-dev...EAP_CD_17_CR1) => 

`
There isn’t anything to compare.
eap-cd-dev and EAP_CD_17_CR1 are identical.
`

we should be OK once this change is merged. Sorry for the turmoil (didn't inspect in that too much detail yesterday, when creating the PR).

@hmlnarik -^ jFYI

**Comment by luck3y** [comment]
@iankko I made this change seperately (and for the stable branches as well while changing jboss-eap-modules and the openshift repos as well. So this can probably be closed now.

**Comment by iankko** [comment]
> @iankko I made this change seperately (and for the stable branches as well while changing jboss-eap-modules and the openshift repos as well. So this can probably be closed now.

@luck3y Wonderful, TYVM! (JFTR, the corresponding JIRA is [CLOUD-3337](https://issues.jboss.org/browse/CLOUD-3337)).

**Pull request closed without merging**
